solr/solr-ref-guide/modules/deployment-guide/pages/basic-authentication-plugin.adoc

```diff
@@ -81,6 +81,14 @@ If `realm` is not defined, it will default to `solr`.
 If you are using SolrCloud, you must upload `security.json` to ZooKeeper.
 An example command and more information about securing your setup can be found at xref:authentication-and-authorization-plugins#in-a-solrcloud-cluster[Authentication and Authorization Plugins In a SolrCloud Cluster].
 
+=== Password Encoding
+
+Solr stores the passwords in the format: `base64(sha256(sha256(salt+password))) base64(salt)`.  
+
+If you edit `security.json` directly then you need to encode the password yourself.  
+You can visit https://clemente-biondo.github.io/ to use a simple web utility that does the encoding for you.
```

Comment:
Dangerous to link to 3rd party tool, we cannot know if his account gets compromised and the script modified. Would rather copy the JS and use it somewhere in our own ref-guide or website, or perhaps even as part of Solr Admin?

Reply:
I once had a client that couldn't figure out how to give me access to their solr admin ui. After multiple failed attempts and clear intent that I *should* have access, I looked up the code in Solr, executed it in an IntelliJ groovy console and generated myself a password, which I then added it to their (unsecured) zookeeper to gain access. (and advised them of what I had done and the need to secure things further). My point is, if we are going to document this, we could easily expose a tool based on our code rather than reference a 3rd party.